vyper/builtins/functions.py

```diff
@@ -305,7 +305,7 @@ def fetch_call_return(self, node):
 
         arg = node.args[0]
         start_expr = node.args[1]
-        length_expr = node.args[2]
+        length_expr = node.args[2].reduced()
```

Comment:
shouldn't we reduce also the `start_expr`?

Reply:
I don't think we need compile time resolution of it

Reply:
what about this path?

https://github.com/tserg/vyper/blob/4ba0b10601f7259bfa76dd4683cc2f4613b314ad/vyper/builtins/functions.py#L326

Reply:
we need the length to be a constant for type-checking purposes (https://github.com/tserg/vyper/blob/4ba0b10601f7259bfa76dd4683cc2f4613b314ad/vyper/builtins/functions.py#L238). it's ok if start is not

Reply:
i get that but if we fold start we can can additional comp-time checks, right (as per the path I shared)?

Reply:
yes, but i wouldn't say they are that important to the working of the compiler. (they should get found as static assertion violations during optimization anyway)